### DISCUSSION TITLE
My minestom server can't be joined from outside of the network

### DISCUSSION THREAD

**Discussion by Flamgop** [Minestom/Minestom#882] [Help] [answered]
Whenever I try joining my minestom server from my external IP it responds with `finishConnect(...) failed: Connection refused`. I have tried a Purpur server with my external IP  (and had a friend connect) and it works fine (same port), I can also join my minestom server from localhost. I don't understand what could be causing this. There are no errors in my server's console.

**Comment by TheMode** [comment]
Can you include the code responsible for starting the server?

**Reply by Flamgop** [reply]
Here's a stripped version of it
```java
MinecraftServer server = MinecraftServer.init();
Properties properties = getOrCreateProperties(); // just creates a properties file with some values (port, ip, motd, max players, seed, render dist and op-level)

// get values from properties
int port = ...
String ip = ...
// etc etc

// register instances & commands
// register scheduled tasks

MojangAuth.init();
OptifineSupport.enable();

server.start(ip, port);
```
I am not using any proxies btw, just attempting to join directly to the server.

**Reply by TheMode** [reply]
You are hiding the IP and port. Try using ipv4/6 wildcard

**Comment by emortaldev** [comment, marked as answer]
Try 0.0.0.0 or 127.0.0.1, "localhost" doesn't allow connections from outside afaik

**Reply by Flamgop** [reply]
Oh, wonder why that is

**Reply by Eoghanmc22** [reply]
iirc 0.0.0.0 is bind on all avaible ips where local host or another ip binds to that ip specifically
